ray_lightning/ray_ddp.py

```diff
@@ -116,10 +121,15 @@ def __init__(self,
             parallel_devices=[],
             cluster_environment=RayEnvironment(world_size=num_workers),
             **ddp_kwargs)
+        resources_per_worker = resources_per_worker if resources_per_worker \
+            else {}
         self.nickname = "ddp_ray"
         self.num_workers = num_workers
-        self.num_cpus_per_worker = num_cpus_per_worker
-        self.use_gpu = use_gpu
+        self.num_cpus_per_worker = resources_per_worker.pop(
+            "CPU", num_cpus_per_worker)
+        self.use_gpu = resources_per_worker.pop(
+            "GPU", 0) > 0 if "GPU" in resources_per_worker else use_gpu
```

Comment:
What's the intended behavior if the user sets `GPU` to something other than 1? Docs indicate that this should override, but the implementation will set it to 1.

Reply:
Changed it so that it can support multiple GPUs per worker if explicitly specified.